webui/src/js/utils/k8s-domain-resource-generator.js

```diff
@@ -353,6 +369,21 @@ define(['models/wkt-project', 'utils/k8s-domain-configmap-generator', 'js-yaml',
       return foundValue ? resources : null;
     }
 
+    function _getAffinityForServerPod(weight) {
+      const podAffinityTerm = {};
+      podAffinityTerm.labelSelector = {};
+      const keyObject = {'key': 'weblogic.cluster.name', 'operator':'In', 'values':['$(CLUSTER_NAME)']};
+      podAffinityTerm.labelSelector.matchExpressions = [keyObject];
+      podAffinityTerm.topologyKey = 'kubernetes.io/hostname';
+      const podObject = {'weight': weight, podAffinityTerm};
+
+      const preferredDuringSchedulingIgnoredDuringExecution = [podObject];
+      const preferred = {preferredDuringSchedulingIgnoredDuringExecution};
+      return  {'podAntiAffinity': preferred};
+    }
+
```

Comment:
This seems like a lot of work for returning a predefined object.  There is no need to build the object up like in Java so why not just do something like:

```
return {
  podAntiAffinity: {
    preferredDuringSchedulingIgnoredDuringExecution: [{
      weight: 100
      podAffinityTerm: {
        topologyKey: 'kubernetes.io/hostname'
        labelSelector: {
          matchExpressions: [{
            key: 'weblogic.clusterName'
            operator: 'In'
            values: [{ '$(CLUSTER_NAME)' }]
          }]
        }
      }
    }]
  }
};
```

Reply:
reworked code per above